massa-execution-worker/src/interface_impl.rs

```diff
@@ -315,6 +332,22 @@ impl Interface for InterfaceImpl {
         }
     }
 
+    fn get_keys_wasmv1(&self, prefix: &[u8], address: Option<String>) -> Result<BTreeSet<Vec<u8>>> {
+        let context = context_guard!(self);
+        let address = match address {
+            Some(address) => massa_models::address::Address::from_str(&address)?,
+            None => context.get_current_address()?,
+        };
+        match (context.get_keys(&address), prefix) {
+            (Some(mut value), prefix) if !prefix.is_empty() => {
+                value.retain(|key| key.iter().zip(prefix.iter()).all(|(k, p)| k == p));
```

Comment:
That's not what its supposed to do is it? Unless I have misunderstood the prefix behaviour. Because here you're not treating it as a prefix since you match the entire key with it.

Reply:
The zip operation yields None if the prefix.iter() yields None, so it only checks that the key and prefix are equal for the length of the prefix.